packages/module-utils/test/generate-dts.spec.ts

```diff
@@ -247,6 +276,75 @@ describe('Generate DTS', function () {
                 "Type 'true' is not assignable to type 'string | undefined'"
             );
         });
+
+        describe('state overrides', () => {
+            it('should support states overridden through a local extend', () => {
+                tk.populate({
+                    'test.st.css':
+                        '.base { -st-states: state1; } .test { -st-states: state1(string); -st-extends: base; }',
+                    'test.ts': `
+                        import { eq } from "./test-kit";
+                        import { st, classes } from "./test.st.css";
+                        
+                        eq<string>(st(classes.test, { state1: 'hello' }));
+                    `,
+                });
+
+                expect(tk.typecheck('test.ts')).to.equal('');
+            });
+
+            it('should warn about incorrect state value that is overridden through a local extend', () => {
+                tk.populate({
+                    'test.st.css':
+                        '.base { -st-states: state1; } .test { -st-states: state1(string); -st-extends: base; }',
+                    'test.ts': `
+                        import { eq } from "./test-kit";
+                        import { st, classes } from "./test.st.css";
+                        
+                        eq<string>(st(classes.test, { state1: true }));
+                    `,
+                });
+
+                expect(tk.typecheck('test.ts')).to.include(
+                    `Type 'true' is not assignable to type 'string | undefined'`
+                );
+            });
+
+            it('should not warn about the same state that is being overridden through a local extend to the same type', () => {
+                tk.populate({
+                    'test.st.css':
+                        '.base { -st-states: state1(string); } .test { -st-states: state1(string); -st-extends: base; }',
+                    'test.ts': `
+                        import { eq } from "./test-kit";
+                        import { st, classes } from "./test.st.css";
+                        
+                        eq<string>(st(classes.test, { state1: 'hello' }));
+                    `,
+                });
+
+                expect(tk.typecheck('test.ts')).to.equal('');
+            });
+
+            it('should warn about state override from imported extend', () => {
+                // inherited state overrides only work in the same stylesheet
+                tk.populate({
+                    'base.st.css': '.part {stStates: state1}',
+                    'test.st.css': `
+                        @st-import [part] from './base.st.css';
+                        .test { -st-extends: part; }`,
+                    'test.ts': `
+                        import { eq } from "./test-kit";
+                        import { st, classes } from "./test.st.css";
+                        
+                        eq<string>(st(classes.test, { state1: true }));
```

Comment:
I'm not sure that states from imports should be visible in the extending stylesheet